internal/resolution/entitysources/catalogdsource.go

```diff
@@ -0,0 +1,159 @@
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	catalogd "github.com/operator-framework/catalogd/pkg/apis/core/v1beta1"
+)
+
+// catalogdEntitySource is a source for(/collection of) deppy defined input.Entity, built from content
+// made accessible on-cluster by https://github.com/operator-framework/catalogd.
+// It is an implementation of deppy defined input.EntitySource
+type catalogdEntitySource struct {
+	client client.Client
+}
+
+// type catalogdEntitySource map[deppy.Identifier]input.Entity
+
+func NewCatalogdEntitySource(ctx context.Context, client client.Client) catalogdEntitySource {
+
+	return catalogdEntitySource{client: client}
+}
+
+func (es catalogdEntitySource) Get(ctx context.Context, id deppy.Identifier) *input.Entity {
+	panic("not implemented")
+}
+
+func (es catalogdEntitySource) Filter(ctx context.Context, filter input.Predicate) (input.EntityList, error) {
+	resultSet := input.EntityList{}
+	entities, err := getEntities(ctx, es.client)
+	if err != nil {
+		return nil, err
+	}
+	for _, entity := range entities {
+		if filter(&entity) {
+			resultSet = append(resultSet, entity)
+		}
+	}
+	return resultSet, nil
+}
+
+func (es catalogdEntitySource) GroupBy(ctx context.Context, fn input.GroupByFunction) (input.EntityListMap, error) {
+	entities, err := getEntities(ctx, es.client)
+	if err != nil {
+		return nil, err
+	}
+	resultSet := input.EntityListMap{}
+	for _, entity := range entities {
+		keys := fn(&entity)
+		for _, key := range keys {
+			resultSet[key] = append(resultSet[key], entity)
+		}
+	}
+	return resultSet, nil
+}
+
+func (es catalogdEntitySource) Iterate(ctx context.Context, fn input.IteratorFunction) error {
+	entities, err := getEntities(ctx, es.client)
+	if err != nil {
+		return err
+	}
+	for _, entity := range entities {
+		if err := fn(&entity); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func jsonMarshal(p interface{}) ([]byte, error) {
+	buf := &bytes.Buffer{}
+	dec := json.NewEncoder(buf)
+	dec.SetEscapeHTML(false)
+	err := dec.Encode(p)
+	if err != nil {
+		return nil, err
+	}
+	out := &bytes.Buffer{}
+	if err := json.Compact(out, buf.Bytes()); err != nil {
+		return nil, err
+	}
+	return out.Bytes(), nil
+}
+
+func getEntities(ctx context.Context, client client.Client) (input.EntityList, error) {
+	entities := input.EntityList{}
+	bundleMetadatas, packageMetdatas, err := fetchMetadata(ctx, client)
+	if err != nil {
+		return nil, err
+	}
+	for _, bundle := range bundleMetadatas.Items {
+		props := map[string]string{}
+
+		for _, prop := range bundle.Spec.Properties {
+			switch prop.Type {
+			case property.TypePackage:
+				type pkg struct {
+					Name    string `json:"packageName"`
+					Version string `json:"version"`
+				}
+				tmp := pkg{}
+				if err := json.Unmarshal(prop.Value, &tmp); err != nil {
+					return nil, err
+				}
+				pkgValue, err := jsonMarshal(tmp)
+				if err != nil {
+					return nil, err
+				}
+				props[property.TypePackage] = string(pkgValue)
```

Comment:
This needs a comment explaining what's going on. To my eye, it's literally unmarshalling and then re-marshaling, which seems like a no-op. Why not consolidate this down to:
```go
case property.TypePackage:
	props[property.TypePackage] = string(prop.Value)
```

Better yet, I'd imagine something more like this:
```go
valsByType := map[string][]json.RawMessage{}
for _, prop := range bundle.Spec.Properties {
	valsByType[prop.Type] = append(valsByType[prop.Type], prop.Value)
}
entityProps := map[string]string{}
for typ, vals := range valsByType {
	switch typ {
	case property.TypePackage:
		// We expect exactly on "olm.package" property
		if len(vals) != 1 { return err }
		entityProps[typ] = string(vals[0])
	default:
		// All other properties can have 0 or more values, so always use a list
		entityProps[typ] = fmt.Sprintf("[%s]", bytes.Join(vals, []byte(",")))
	}
}
```